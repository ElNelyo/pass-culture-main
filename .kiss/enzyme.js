// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
