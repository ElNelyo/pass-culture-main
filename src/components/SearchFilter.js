/* eslint-disable */
import PropTypes from 'prop-types'
import moment from 'moment'
import React from 'react'

export const SearchFilter = ({
  handleClearQueryParams,
  handleQueryParamsChange,
}) => (
  <div className="search-filter">
    <form />
    <div>
      <h2>DATE</h2>
      <div className="field checkbox">
        <label id="from_date" className="label">
          Tout de suite !
        </label>
        <input
          id="from_date"
          className="input is-normal"
          onChange={() => handleQueryParamsChange({ from_date: moment.now() })}
          type="checkbox"
        />
      </div>
      <div className="field checkbox">
        <label id="from_date" className="label">
          Entre 1 et 5 jours
        </label>
        <input
          id="from_date"
          className="input is-normal"
          onChange={() => handleQueryParamsChange({ from_date: moment.now() })}
          type="checkbox"
        />
      </div>
    </div>
    <div>
      <h2>DISTANCE</h2>
      <select
        name="select"
        className="select"
        onChange={() => handleQueryParamsChange({ distance: 'value' })}>
        {/*  { distance: 1 } */}
        <option value="1">Moins d'1 km</option>
        <option value="2">Moins de 10 km</option>
        <option value="3">Moins de 50 km</option>
        <option value="4">Toutes distances</option>
      </select>
    </div>
    <div>
      <h2>QUOI</h2>
      <div className="field checkbox">
        <label id="from_date" className="label">
          Applaudir
        </label>
        <input
          id="from_date"
          className="input is-normal"
          onChange={() => handleQueryParamsChange({ from_date: moment.now() })}
          type="checkbox"
        />
      </div>
    </div>

    <button className="button" type="button" onClick={handleClearQueryParams}>
      RÃ©initialiser
    </button>
    <button className="button" type="submit">
      Filtrer
    </button>
  </div>
)

SearchFilter.propTypes = {
  handleClearQueryParams: PropTypes.func.isRequired,
  handleQueryParamsChange: PropTypes.func.isRequired,
}
