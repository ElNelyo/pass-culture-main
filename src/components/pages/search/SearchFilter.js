/* eslint-disable */
import get from 'lodash.get'
import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { shallowCompare } from 'react-redux'

import FilterByDates from './FilterByDates'
import FilterByDistance from './FilterByDistance'
import FilterByOfferTypes from './FilterByOfferTypes'

import { INITIAL_FILTER_PARAMS, searchFiltersAdded } from '../search/utils'

function getFirstChangingKey(previousObject, nextObject) {
  return Object.keys(nextObject).find(key => {
    const isNewFalsy = nextObject[key] === null || nextObject[key] === ''
    const isPreviousFalsy =
      typeof previousObject[key] === 'undefined' ||
      previousObject[key] === null ||
      previousObject === ''
    if (isNewFalsy && isPreviousFalsy) {
      return false
    }
    return previousObject[key] !== nextObject[key]
  })
}

class SearchFilter extends Component {
  constructor(props) {
    super(props)
    this.state = {
      filterParams: Object.assign({}, props.queryParams),
      isNewFilter: false,
    }
  }

  componentDidUpdate(prevProps) {
    const { queryParams } = this.props
    // TODO: eslint does not support setState inside componentDidUpdate
    if (queryParams !== prevProps.queryParams) {
      this.setState({
        filterParams: queryParams,
        isNewFilter: false,
      })
    }
  }

  onFilterClick = () => {
    const { handleQueryParamsChange } = this.props
    const { filterParams, isNewFilter } = this.state

    handleQueryParamsChange(filterParams, {
      isRefreshing: isNewFilter,
      pathname: '/recherche/resultats',
    })
  }

  onResetClick = () => {
    this.setState({
      filterParams: INITIAL_FILTER_PARAMS,
      isNewFilter: getFirstChangingKey(
        this.props.queryParams,
        INITIAL_FILTER_PARAMS
      ),
    })
  }

  handleFilterParamsChange = (newValue, callback) => {
    const { queryParams } = this.props
    const { filterParams } = this.state

    const nextFilterParams = Object.assign({}, filterParams, newValue)

    const isNewFilter = getFirstChangingKey(queryParams, newValue)

    this.setState(
      {
        filterParams: nextFilterParams,
        isNewFilter,
      },
      callback
    )
  }

  handleFilterParamAdd = (key, value, callback) => {
    const { filterParams } = this.state

    const encodedValue = encodeURI(value)
    let nextValue = encodedValue
    const previousValue = filterParams[key]
    if (get(previousValue, 'length')) {
      const args = previousValue.split(',').concat([encodedValue])
      args.sort()
      nextValue = args.join(',')
    }

    this.handleFilterParamsChange({ [key]: nextValue }, callback)
  }

  handleFilterParamRemove = (key, value, callback) => {
    const { filterParams } = this.state

    const previousValue = filterParams[key]

    if (get(previousValue, 'length')) {
      const encodedValue = encodeURI(value)
      let nextValue = previousValue
        .replace(`,${encodedValue}`, '')
        .replace(encodedValue, '')
      if (nextValue[0] === ',') {
        nextValue = nextValue.slice(1)
      }
      this.handleFilterParamsChange({ [key]: nextValue }, callback)
    }
  }

  render() {
    const { handleClearQueryParams } = this.props
    const { filterParams, isNewFilter } = this.state

    const isNullFilter = searchFiltersAdded(INITIAL_FILTER_PARAMS, filterParams)

    return (
      <div id="search-filter-menu">
        <FilterByDates
          handleFilterParamsChange={this.handleFilterParamsChange}
          handleFilterParamAdd={this.handleFilterParamAdd}
          handleFilterParamRemove={this.handleFilterParamRemove}
          filterParams={filterParams}
          title="QUAND"
        />
        <FilterByDistance
          handleFilterParamsChange={this.handleFilterParamsChange}
          filterParams={filterParams}
          title="OÙ"
        />
        <FilterByOfferTypes
          handleFilterParamAdd={this.handleFilterParamAdd}
          handleFilterParamRemove={this.handleFilterParamRemove}
          filterParams={filterParams}
          title="QUOI"
        />
        <button
          className="button fs24"
          disabled={isNullFilter}
          onClick={this.onResetClick}
          type="button">
          Réinitialiser
        </button>
        <button
          className="button fs24"
          disabled={!isNewFilter}
          onClick={this.onFilterClick}
          type="button">
          Filtrer
        </button>
      </div>
    )
  }
}

SearchFilter.propTypes = {
  handleQueryParamsChange: PropTypes.func.isRequired,
  queryParams: PropTypes.object.isRequired,
}

export default SearchFilter
