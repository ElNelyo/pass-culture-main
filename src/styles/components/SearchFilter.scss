.search-filter {
  background-color: $grey-light;
  border-radius: 0.25em;

  h2 {
    font-style: italic;
    margin-bottom: 1em;
    margin-top: 1em;
    text-transform: uppercase;
    text-align: center;
  }

  div.field.checkbox {
    border-color: $primary;
  }

  button {
    background-color: $grey-light;
    border-color: white;
    border-radius: 0em;
  }
}
