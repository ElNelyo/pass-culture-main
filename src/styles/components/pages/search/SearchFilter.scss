#search-filter-menu {
  background-color: $grey-lighter;
  border-radius: 0 0 0.25em 0.25em;
  // left: 0.5rem;
  // right: 0.5rem;

  // h2 {
  //   margin-bottom: 1em;
  //   margin-top: 1em;
  // }

  button {
    background-color: $grey-lighter;
    border-color: $white;
    border-radius: 0;
    color: $primary;
    font-size: 1.375rem;
  }
}

#filter-by-dates {
  .list .item {
    border-radius: 4px;
    border: solid $primary 1px;
  }

  .list .item + .item {
    @extend .ml22;
  }

  .list .item .field {
    margin-bottom: 0;
  }

  .list .item .field .input {
    margin-right: 7px;
  }
}

/*
#filter-by-distance {
  padding-bottom: 0.25rem;

  select {
    border-color: $primary;
    border-radius: $control-radius;
    min-width: 250px;
    margin: 3rem;
    padding: 1rem;
    -webkit-appearance: none;
    -moz-appearance: none;
    background: url('/icons/dropdown-disclosure-down.svg') no-repeat;
    background-position: right;
  }
}
*/

#filter-by-offer-types {
  label {
    color: $white;
    height: 23px;
    width: 86px;
    opacity: 0.8;
    background-color: #000000;
  }

  margin-bottom: 1rem;
}

.filter-menu-outer {
  display: flex; /* displays flex-items (children) inline */
  overflow-x: auto;
}

.filter-menu-inner {
  // flex: 0 0 25%;

  /* doesn't grow nor shrink, initial width set to 25% of the parent's */
  padding: 0.5em;
  border-radius: 1.125em;

  img {
    height: 86px;
  }
}
