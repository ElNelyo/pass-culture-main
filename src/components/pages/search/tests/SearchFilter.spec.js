import React from 'react'
import { shallow } from 'enzyme'

import SearchFilter from '../SearchFilter'

describe('src | components | pages | search | SearchFilter', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        handleQueryParamsChange: jest.fn(),
        queryParams: {},
      }

      // when
      const wrapper = shallow(<SearchFilter {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
