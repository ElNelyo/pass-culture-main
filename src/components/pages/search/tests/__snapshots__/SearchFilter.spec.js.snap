// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | search | SearchFilter snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchFilter
    handleQueryParamsChange={[MockFunction]}
    queryParams={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FilterByDates
          filterParams={Object {}}
          handleFilterParamAdd={[Function]}
          handleFilterParamRemove={[Function]}
          handleFilterParamsChange={[Function]}
          title="QUAND"
        />,
        <Connect(FilterByDistance)
          filterParams={Object {}}
          handleFilterParamsChange={[Function]}
          title="OÙ"
        />,
        <Connect(FilterByOfferTypes)
          filterParams={Object {}}
          handleFilterParamAdd={[Function]}
          handleFilterParamRemove={[Function]}
          title="QUOI"
        />,
        <button
          className="button fs24"
          onClick={[Function]}
          type="button"
        >
          Réinitialiser
        </button>,
        <button
          className="button fs24"
          onClick={[Function]}
          type="button"
        >
          Filtrer
        </button>,
      ],
      "id": "search-filter-menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filterParams": Object {},
          "handleFilterParamAdd": [Function],
          "handleFilterParamRemove": [Function],
          "handleFilterParamsChange": [Function],
          "title": "QUAND",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filterParams": Object {},
          "handleFilterParamsChange": [Function],
          "title": "OÙ",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filterParams": Object {},
          "handleFilterParamAdd": [Function],
          "handleFilterParamRemove": [Function],
          "title": "QUOI",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Réinitialiser",
          "className": "button fs24",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Réinitialiser",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Filtrer",
          "className": "button fs24",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Filtrer",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FilterByDates
            filterParams={Object {}}
            handleFilterParamAdd={[Function]}
            handleFilterParamRemove={[Function]}
            handleFilterParamsChange={[Function]}
            title="QUAND"
          />,
          <Connect(FilterByDistance)
            filterParams={Object {}}
            handleFilterParamsChange={[Function]}
            title="OÙ"
          />,
          <Connect(FilterByOfferTypes)
            filterParams={Object {}}
            handleFilterParamAdd={[Function]}
            handleFilterParamRemove={[Function]}
            title="QUOI"
          />,
          <button
            className="button fs24"
            onClick={[Function]}
            type="button"
          >
            Réinitialiser
          </button>,
          <button
            className="button fs24"
            onClick={[Function]}
            type="button"
          >
            Filtrer
          </button>,
        ],
        "id": "search-filter-menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterParams": Object {},
            "handleFilterParamAdd": [Function],
            "handleFilterParamRemove": [Function],
            "handleFilterParamsChange": [Function],
            "title": "QUAND",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterParams": Object {},
            "handleFilterParamsChange": [Function],
            "title": "OÙ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterParams": Object {},
            "handleFilterParamAdd": [Function],
            "handleFilterParamRemove": [Function],
            "title": "QUOI",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Réinitialiser",
            "className": "button fs24",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Réinitialiser",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Filtrer",
            "className": "button fs24",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Filtrer",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
